x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java

```diff
@@ -134,12 +137,17 @@ private void reindexDataframeAndStartAnalysis(DataFrameAnalyticsTask task, DataF
 
         // Refresh to ensure copied index is fully searchable
         ActionListener<BulkByScrollResponse> reindexCompletedListener = ActionListener.wrap(
-            bulkResponse ->
+            bulkResponse -> {
+                if (task.isStopping()) {
+                    return;
```

Comment:
might be helpful for debugging to add a `debug` log here. I imagine it would be useful to figure out race conditions/CI failures.